apps/web/src/components/Basenames/UsernameProfileSectionFrames/FrameTheme.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @next/next/no-img-element */
+/* eslint-disable react/prop-types */
+import type { FrameUIComponents, FrameUITheme } from '@frames.js/render/ui';
+import classNames from 'classnames';
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+import baseLoading from './base-loading.gif';
+
+type StylingProps = {
+  className?: string;
+  style?: React.CSSProperties;
+};
+
+export const theme: FrameUITheme<StylingProps> = {
+  Error: {
+    className:
+      'flex flex-col rounded-lg overflow-hidden bg-transparent relative items-center justify-center opacity-50',
+  },
+  Root: {
+    className:
+      'relative flex flex-col rounded-lg items-center justify-center overflow-hidden bg-transparent relative border border-palette-line/20 min-w-[245px] min-h-[245px]',
+  },
+  ButtonsContainer: {
+    className:
+      'w-full text-xs sm:text-sm flex p-2 justify-around bg-palette-positiveForeground gap-2 sm:gap-4',
+  },
+  Button: {
+    className:
+      'text-sm grow py-3 px-5 rounded-full bg-[#F3F3F3] text-palette-secondaryForeground font-medium transition-colors hover:bg-state-b-hovered min-h-[44px]',
+  },
+  ImageContainer: {
+    className: 'flex flex-grow border-0 overflow-hidden',
+  },
+  TextInputContainer: {
+    className:
+      'bg-palette-positiveForeground flex flex-row items-center justify-around gap-2 w-full',
+  },
+  TextInput: {
+    className: 'rounded-xl border border-palette-line/20 px-3 py-2 mt-3 mx-3 w-full',
+  },
+};
+
+type TransitionWrapperProps = {
+  aspectRatio: '1:1' | '1.91:1';
+  src: string;
+  alt: string;
+  onImageLoadEnd: () => void;
+  stylingProps: StylingProps;
+  status: 'frame-loading' | 'frame-loading-complete';
+};
+
+function TransitionWrapper({
+  aspectRatio,
+  src,
+  alt,
+  onImageLoadEnd,
+  stylingProps,
+  status,
+}: TransitionWrapperProps) {
+  const [isTransitioning, setIsTransitioning] = useState(true);
+  const isLoading = status === 'frame-loading';
+  useEffect(() => {
+    let timeout: number;
+    if (!isLoading) {
+      timeout = setTimeout(() => {
+        setIsTransitioning(false);
+      }, 500) as unknown as number;
+    } else {
+      setIsTransitioning(true);
+    }
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [isLoading]);
+
+  const ar = aspectRatio.replace(':', '/');
```

Comment:
Any reason to do this instead of just writing the aspect ratios everywhere as '1/1', '1.91/1'?

Reply:
sadly we don't control this--it's an internal frames.js thing that needs to be manually changed for css applications